### ISSUE TITLE
Cannot access index in properties of dynamic object

### ISSUE THREAD

**Issue by NielsUll** [davideicardi/DynamicExpresso#92]
When trying to access vlues of an array (or dictionary) inside a dynamic object, I get the error:

DynamicExpresso.Exceptions.ParseException : No applicable indexer exists in type 'Object' (at index 7).

```
    [Test]
    public void Get_value_of_a_nested_array()
   {
       dynamic dyn = new ExpandoObject();
       dyn.Sub = new int[] {42};
       var interpreter = new Interpreter().SetVariable("dyn", (object)dyn);
       Assert.AreEqual(dyn.Sub[0], interpreter.Eval("dyn.Sub[0]"));
}
```
 This is also mentioned on Stack Overflow, https://stackoverflow.com/questions/53040955/how-can-i-use-dynamic-variables-with-anonymous-types

**Comment by davideicardi** [comment]
Thanks @NielsUll Not sure when I can work on it, so any help is appreciated.

**Comment by davideicardi** [comment]
Closed by #142
